### PR TITLE
Fix comment and bump version, add dependencies

### LINKED ISSUE #1081: Python send_with_params ignores expiration
## Description

Python send_with_params() ignores expiration field

### LINKED ISSUE #1252: Make pip install automatically install required dependencies
## Description

Make pip install automatically install required dependencies like dacite and pyhumps

Can maybe done in some way by adding them to pyproject.toml 

There was https://github.com/iotaledger/iota-sdk/pull/830 already, but this didn't seem to fix it

## Motivation

Don't return errors when users don't have this installed

## Are you planning to do it yourself in a pull request?

No.

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Fix comment and bump version, add dependencies

## Links to any relevant issues

Closes #1081 
Fixes https://github.com/iotaledger/iota-sdk/issues/1252

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Documentation Fix

## How the change has been tested

Tested workflow in my fork and installation of the wheel with docker worked then

### CODE REVIEW

File: bindings/python/pyproject.toml [hunk 1]
```diff
@@ -4,6 +4,7 @@ build-backend = "maturin"
 
 [project]
 name = "iota-sdk"
+dependencies = ["dacite >= 1.8.1", "pyhumps >= 3.8.0"]
```
Comment (thibault-martinez) [inline bindings/python/pyproject.toml:7]: What's the difference with those in `requirements-dev.txt`?

Comment (Thoralf-M) [reply bindings/python/pyproject.toml:7]: This should also be included in the final package, the requirements-dev.txt not
But have to test first if it actually works this way

Comment (Thoralf-M) [reply bindings/python/pyproject.toml:7]: Works

Comment (thibault-martinez) [reply bindings/python/pyproject.toml:7]: And these are the only required deps ?

**Review by thibault-martinez — APPROVED** [review]

File: bindings/python/pyproject.toml (hunk 1 above)
Comment (Thoralf-M) [reply bindings/python/pyproject.toml:7]: Yes

**Review by kwek20 — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**